pymongo/asynchronous/client_session.py

```diff
@@ -598,7 +601,10 @@ def _inherit_option(self, name: str, val: _T) -> _T:
 
     async def with_transaction(
         self,
-        callback: Callable[[AsyncClientSession], _T],
+        callback: Union[
+            Callable[[AsyncClientSession], Coroutine[Any, Any, _T]],
```

Comment:
Should we use Coroutine here or Awaitable? Which is a better fit?

Reply:
We've used `Coroutine` consistently so far, `Awaitable` is a less-specific type hint that allows several other objects besides asynchronous functions.